packages/calcite-components/src/components/action-bar/action-bar.tsx

```diff
@@ -342,8 +342,32 @@ export class ActionBar
       expanded,
       overflowCount,
     });
+
+    this.removeOverflowGroups(height, actionGroups, layout);
   }, overflowActionsDebounceInMs);
 
+  private removeOverflowGroups(
+    height: number,
+    actionGroups: HTMLCalciteActionGroupElement[],
+    layout: "horizontal" | "vertical",
+  ) {
+    let groupHeight = 0;
+    if (layout === "vertical") {
+      actionGroups.forEach((group) => {
+        groupHeight += group.clientHeight;
+      });
+      while (height > groupHeight || Math.abs(height - groupHeight) <= 20) {
+        const group = actionGroups.shift();
+        group.style.display = "none";
```

Comment:
question: why are groups being hidden?

Reply:
I thought that's what we agreed when we all met to discuss this issue. We agreed to remove action groups when the vertical action bar was too long.